cmd/libsnap-confine-private/snap.c

```diff
@@ -97,6 +97,108 @@ static int skip_one_char(const char **p, char c)
 	return 0;
 }
 
+void sc_instance_name_validate(const char *instance_name,
+			       struct sc_error **errorp)
+{
+	struct sc_error *err = NULL;
+
+	// Ensure that name is not NULL
+	if (instance_name == NULL) {
+		err =
+		    sc_error_init(SC_SNAP_DOMAIN, SC_SNAP_INVALID_INSTANCE_NAME,
+				  "snap instance name cannot be NULL");
+		goto out;
+	}
+
+	const char *instance_sep = strchr(instance_name, '_');
```

Comment:
Would it be sane to split the instance name into snap name and instance key and just validate both?

Reply:
See my comments about using strsep from https://github.com/snapcore/snapd/pull/5584#discussion_r206557265 - I also wonder if this and the snap-update-ns bootstrap code can be somehow unified?

Reply:
Yes, you redid the implementation in PR 5584 with some nice strsep() cleanups, it would be great to have the equivalent changes here.